docs/english/concepts/ai-apps.md

```diff
@@ -1,80 +1,146 @@
-  // threadContextChanged is optional
-  // If you use your own optional threadContextStore you likely won't use it
-  threadContextChanged: async ({ say, setStatus, setSuggestedPrompts, setTitle }) => {},
-  userMessage: async ({ say, getThreadContext, setStatus, setSuggestedPrompts, setTitle }) => {},
+  /**
+   * (Recommended) A custom ThreadContextStore can be provided, inclusive of methods to
+   * get and save thread context. When provided, these methods will override the `getThreadContext`
+   * and `saveThreadContext` utilities that are made available in other Assistant event listeners.
+   */
+  // threadContextStore: {
+  //   get: async ({ context, client, payload }) => {},
+  //   save: async ({ context, client, payload }) => {},
+  // },
+
+  /**
+   * `assistant_thread_started` is sent when a user opens the Assistant container.
+   * This can happen via DM with the app or as a side-container within a channel.
+   */
+  threadStarted: async ({ event, logger, say, setSuggestedPrompts, saveThreadContext }) => {},
+  
+  /**
+   * `assistant_thread_context_changed` is sent when a user switches channels
+   * while the Assistant container is open. If `threadContextChanged` is not
+   * provided, context will be saved using the AssistantContextStore's `save`
+   * method (either the DefaultAssistantContextStore or custom, if provided).
+   */
+  threadContextChanged: async ({ logger, saveThreadContext }) => {},
+  
+  /**
+   * Messages sent from the user to the Assistant are handled in this listener.
+   */
+  userMessage: async ({ client, context, logger, message, getThreadContext, say, setTitle, setStatus }) => {},
 });
 ```
 
+:::info[Consider the following]
+You _could_ go it alone and [listen](/tools/bolt-js/concepts/event-listening) for the `assistant_thread_started`, `assistant_thread_context_changed`, and `message.im` events in order to implement the AI features in your app. That being said, using the `Assistant` class will streamline the process. And we already wrote this nice guide for you!
+:::
+
 While the `assistant_thread_started` and `assistant_thread_context_changed` events do provide Slack-client thread context information, the `message.im` event does not. Any subsequent user message events won't contain thread context data. For that reason, Bolt not only provides a way to store thread context — the `threadContextStore` property — but it also provides a `DefaultThreadContextStore` instance that is utilized by default. This implementation relies on storing and retrieving [message metadata](/messaging/message-metadata/) as the user interacts with the app. 
 
 If you do provide your own `threadContextStore` property, it must feature `get` and `save` methods.
 
 :::tip[Be sure to give the [reference docs](/tools/bolt-js/reference#agents--assistants) a look!]
 :::
 
-## Handling a new thread {#handling-new-thread}
+### Configuring your app to support the `Assistant` class
```

Comment:
⭐ praise: These lists are so useful! I find I consult this reference so often in troubleshooting and find thrill here.